### PR TITLE
Optimizes image sizes

### LINKED ISSUE #1106: Ensure images are using `sizes` prop
Need to make sure we’re using correct sizes for good web core vitals and Lighthouse scores.

### PR DESCRIPTION
**Pull request by manovotny**
Optimizes image sizes to address Lighthouse feedback.

CleanShot 2023-08-02 at 15 24 24@2x

Fixes #1106

### CODE REVIEW

File: app/product/[handle]/page.tsx [hunk 1]
```diff
@@ -131,13 +131,13 @@ async function RelatedProducts({ id }: { id: string }) {
                   currencyCode: product.priceRange.maxVariantPrice.currencyCode
                 }}
                 src={product.featuredImage?.url}
-                width={600}
-                height={600}
+                fill
+                sizes="(min-width: 1024px) 20vw, (min-width: 768px) 25vw, (min-width: 640px) 33vw, (min-width: 475px) 50vw, 100vw"
```
Comment (manovotny) [inline app/product/[handle]/page.tsx:135]: This is `product detail page → related products`.

File: components/carousel.tsx [hunk 2]
```diff
@@ -10,27 +10,29 @@ export async function Carousel() {
...
+              <GridTileImage
+                alt={product.title}
+                label={{
+                  title: product.title,
+                  amount: product.priceRange.maxVariantPrice.amount,
+                  currencyCode: product.priceRange.maxVariantPrice.currencyCode
+                }}
+                src={product.featuredImage?.url}
+                fill
+                sizes="(min-width: 1024px) 16vw, (min-width: 768px) 25vw, 50vw"
```
Comment (manovotny) [inline components/carousel.tsx:30]: This is `home page → scrolling product carousel`.

File: components/grid/three-items.tsx [hunk 3]
```diff
@@ -6,13 +6,15 @@ import Link from 'next/link';
...
     >
-      <Link className="block h-full" href={`/product/${item.handle}`}>
+      <Link className="relative block aspect-square h-full w-full" href={`/product/${item.handle}`}>
         <GridTileImage
           src={item.featuredImage.url}
-          width={size === 'full' ? 1080 : 540}
-          height={size === 'full' ? 1080 : 540}
+          fill
+          sizes={
+            size === 'full' ? '(min-width: 768px) 66vw, 100vw' : '(min-width: 768px) 33vw, 100vw'
```
Comment (manovotny) [inline components/grid/three-items.tsx:16]: This is `home page → three large boxes`.

File: components/layout/product-grid-items.tsx [hunk 4]
```diff
@@ -17,8 +17,8 @@ export default function ProductGridItems({ products }: { products: Product[] })
                 currencyCode: product.priceRange.maxVariantPrice.currencyCode
               }}
               src={product.featuredImage?.url}
-              width={600}
-              height={600}
+              fill
+              sizes="(min-width: 768px) 33vw, (min-width: 640px) 50vw, 100vw"
```
Comment (manovotny) [inline components/layout/product-grid-items.tsx:21]: This is `search page → search results`.

File: components/product/gallery.tsx [hunk 5]
```diff
@@ -27,13 +27,13 @@ export function Gallery({ images }: { images: { src: string; altText: string }[]
...
+    <>
+      <div className="relative aspect-square h-full max-h-[550px] w-full overflow-hidden">
         {images[imageIndex] && (
           <Image
-            className="relative h-full w-full object-contain"
-            height={600}
-            width={600}
+            className="h-full w-full object-contain"
+            fill
+            sizes="(min-width: 1024px) 66vw, 100vw"
```
Comment (manovotny) [inline components/product/gallery.tsx:36]: This is `product detail page → enlarged / primary image`.

File: components/product/gallery.tsx [hunk 6]
```diff
@@ -60,33 +60,34 @@ export function Gallery({ images }: { images: { src: string; altText: string }[]
...
+                  aria-label="Enlarge product image"
+                  href={createUrl(pathname, imageSearchParams)}
+                  scroll={false}
+                  className="h-full w-full"
+                >
+                  <GridTileImage
+                    alt={image.altText}
+                    src={image.src}
+                    width={80}
+                    height={80}
```
Comment (manovotny) [inline components/product/gallery.tsx:82]: This is `product detail page → smaller gallery images`.

File: app/product/[handle]/page.tsx [hunk 7]
```diff
@@ -131,13 +131,13 @@ async function RelatedProducts({ id }: { id: string }) {
                   currencyCode: product.priceRange.maxVariantPrice.currencyCode
                 }}
                 src={product.featuredImage?.url}
-                width={600}
-                height={600}
+                fill
```
Comment (styfle) [inline app/product/[handle]/page.tsx:134]: Are you missing `object-fit`?

File: components/carousel.tsx (hunk 2 above)
Comment (styfle) [reply components/carousel.tsx:30]: How did you come up with 16vw? Are there 6 columns at 1024px?

Comment (styfle) [reply components/carousel.tsx:30]: This is a matter of personal preference, but generally I like using `max-width` so the left most is the smallest device and each device is a larger screen as you go to the right, and the fallback is for screen sizes that are super wide.

For example, see the code snippet here: https://nextjs.org/docs/pages/api-reference/components/image#sizes

Comment (manovotny) [reply components/carousel.tsx]: > How did you come up with 16vw? Are there 6 columns at 1024px?

I honestly don't know what I was thinking... I just pushed a commit to do 50vw → 33vw → 25vw.

Comment (manovotny) [reply components/carousel.tsx]: > This is a matter of personal preference, but generally I like using `max-width` so the left most is the smallest device and each device is a larger screen as you go to the right

I can totally do it this way if that is the preference / best practice. This just fit my mental model paired with Tailwind as it operates with `min-width`. Made matching the surrounding styles with `md` and `lg` easier for my brain. 

https://tailwindcss.com/docs/responsive-design

File: app/product/[handle]/page.tsx (hunk 7 above)
Comment (manovotny) [reply app/product/[handle]/page.tsx:134]: No. `object-contain` is *inside* of `GridTileImage`.

**Review by leerob — APPROVED** [review]

**Pull request merged**